components/com_content/src/Model/ArticlesModel.php

```diff
@@ -491,6 +491,23 @@ protected function getListQuery()
             $query->where($authorWhere . $authorAliasWhere);
         }
 
+        // Filter by checked_out status
+        $checkedOut = $this->getState('filter.checked_out', null);
+        if ($checkedOut !== null && $checkedOut !== '' && $checkedOut !== false) {
+            if ($checkedOut === 1) {
+                // Only checked out articles
+                $query->where($db->quoteName('a.checked_out') . ' > 0');
+            } elseif ($checkedOut === 0) {
+                // Only not checked out articles
+                $query->where($db->quoteName('a.checked_out') . ' = 0');
+            } else {
+                // Checked out by specific user
+                $checkedOut = (int) $checkedOut;
```

Comment:
```suggestion
                $checkedOut = (int) $checkedOut;
```
Move after line 496.

Reply:
@QuyTon Sir , $checkedOut = (int) $checkedOut; is now moved to line 497 (right after the condition check on line 496)